Tensile/Source/lib/CMakeLists.txt

```diff
@@ -21,6 +21,8 @@
 
 cmake_minimum_required(VERSION 3.5)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14" )
```

Comment:
Shouldn't this be `set( CMAKE_CXX_STANDARD 14 )` or `set_target_properties( <___> PROPERTIES CXX_STANDARD 14 CXX_STANDARD_REQUIRED ON )`

I'm not saying that this should hold up the merge, just seeking CMake best practices.